### ISSUE TITLE
Integrate new custom layer feature if possible (at least make an example)

### ISSUE THREAD

**Issue by Wykks** [Wykks/ngx-mapbox-gl#82]
https://github.com/mapbox/mapbox-gl-js/pull/7039

**Comment by bryandesigning** [comment]
This feature honestly looks so damn promising, very excited for this

**Comment by bryandesigning** [comment]
Side note - have you made any progress w/ this?

I've got stuff rendering via a custom layer in my project (bypassing ngx-mapbox-gl using the direct map object)

Just trying to figure out the best way to dynamically update custom layers w/ data from my store

**Comment by bryandesigning** [comment]
Also, What if you moved markers into a purely WebGL existence (instead of custom html/css w/ transforms)

This would provide a great performance benefit to larger amounts of markers and could open doors to more cool things such as custom map drawings via coordinates etc.

**Comment by Wykks** [comment]
Not started yet; but it should not be very complicated.

About your comment on markers stuff, it's already there (and it's recommended, html markers are just a fallback when you need to render html stuff like angular component for example).
https://docs.mapbox.com/mapbox-gl-js/style-spec/#layers-symbol

**Comment by bryandesigning** [comment]
By the markers, I meant so you could have click/drag events, etc

**Comment by Wykks** [comment]
Already there too, with mgl-layer with type symbol.

Something like a mgl-gl-marker which internally create a layer with one symbol could be done. But I don't think it's a good idea to do this kind of work.
Simply put, I believe that it's better to embrace the way mapbox-gl layer works instead of fighting it.

**Comment by bryandesigning** [comment]
Do you have discord or slack where I can contact you for a quick chat? I wanna pick your brain for two seconds :)

**Comment by Wykks** [comment]
Yes, although I'm quite busy these days, do I may not be here in real time :smile: 

Wykks#6730 on discord

Also

https://matrix.to/#/@wykks:matrix.org

**Comment by dmytro-gokun** [comment]
@Wykks Have this been implemented completely or do you there some more work to do here? Let me know and i'll see if I can finish it.

**Comment by Wykks** [comment]
Nothing implemented here. You can work on it :)

**Comment by dmytro-gokun** [comment]
@Wykks Thanks for the answer! Okay, i will assign it to myself and look at it when i have time. Hopefully, soon :)

In any case, anyone interested in this feature is welcome to chime in and give thoughts or submit a PR.
